pkg/scheduler/internal/queue/pod_backoff.go

```diff
@@ -91,12 +94,14 @@ func (pbm *PodBackoffMap) ClearPodBackoff(nsPod ktypes.NamespacedName) {
 
 // CleanupPodsCompletesBackingoff execute garbage collection on the pod backoff,
 // i.e, it will remove a pod from the PodBackoffMap if
-// lastUpdateTime + maxBackoffDuration is before the current timestamp
+// lastUpdateTime + maxDuration >> timestamp
+// We should wait longer than the maxDuration so that we don't lose the
+// information when a scheduling cycle is delayed.
 func (pbm *PodBackoffMap) CleanupPodsCompletesBackingoff() {
 	pbm.lock.Lock()
 	defer pbm.lock.Unlock()
 	for pod, value := range pbm.podLastUpdateTime {
-		if value.Add(pbm.maxDuration).Before(time.Now()) {
+		if value.Add(2 * pbm.maxDuration).Before(pbm.clock.Now()) {
```

Comment:
What is the assumption here? why multiplying by 2 addresses the problem? if the pod stayed in the active queue for maxDuration or longer after a backoff, wouldn't we still incorrectly cleanup the pod state?

Reply:
We would. But then we are talking 10s in the active queue, which should be enough for most cases. This should be enough for processing about 1000 pods. But we can always tune this higher (to 10x, maybe?). The point is that just maxDuration is definitely wrong.

There can be better architectural solutions, but this fix should be easy to backport.

Reply:
Sg, I'll add a comment.

There is already an open PR that goes in the direction that you suggest. And I also commented the same thing :) https://github.com/kubernetes/kubernetes/pull/78849#discussion_r371525344

Reply:
Done